### ISSUE TITLE
Can not export pytorch model to onnx.

### ISSUE THREAD

**Issue by kafan1986** [pytorch/pytorch#79483]
### 🐛 Describe the bug

Pytorch model converts OK. But during inference would through below error.

`Fail: [ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while running ConcatFromSequence node. Name:'ConcatFromSequence_2162' Status Message: concat.cc:156 PrepareForCompute Non concat axis dimensions must match: Axis 1 has mismatched dimensions of 12 and 14`

Already raised an issue in Onnx repo and was directed here https://github.com/microsoft/onnxruntime/issues/11672

The model input has variable input length in one of the dimension.

```python
class LengthRegulator(nn.Module):

    def __init__(self):
        super().__init__()

    @staticmethod
    @torch.jit.script_if_tracing
    def onnx_helper(x_expanded: List[torch.Tensor]):
        temp_tensor = torch.stack(x_expanded)
        return temp_tensor

    def forward(self, x: torch.Tensor, dur: torch.Tensor) -> torch.Tensor:
        dur[dur < 0] = 0.
        x_expanded = []
        for i in range(x.size(0)):
            x_exp = torch.repeat_interleave(x[i], (dur[i] + 0.5).long(), dim=0)
            x_expanded.append(x_exp)

        output = self.onnx_helper(x_expanded)
        return output
```

This is the module that is giving issue. I even went with the torch.jit.script_if_tracing to explicitly define the parameter datatype. It removes the warning messages received earlier during the pytorch to onnx conversion step but the same error during inference remains.

```
/home/xxx/TTS_TFLite/ForwardTacotron/models/common_layers.py:20: TracerWarning: torch.Tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  temp_var = torch.Tensor(temp_var)
```

### Versions

Pytorch version: latest nightly version (1.11.0+cu102) 
OnnxRuntime version: latest 1.11.1

**Comment by shubhambhokare1** [comment]
Hi @kafan1986,

We’ve gone ahead and closed this issue because it is stale.
If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.

Thanks,
ONNX Converter team
